app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -401,6 +401,7 @@ def public_key
 
     def public_key=(key)
       attrs[:public_key] = key
+      ca_cert_store(key)
```

Comment:
This is done in line 465, is it actually needed in both?

Reply:
this code adds the new assigned public key to the store

Reply:
This is only relevant for API V3 and we initialize the store in the client initialization.

Reply:
but the code takes a public key and store it, but only in create or new, but what happens when you change the public key?

Reply:
This addition will not help if it's not sent to the client.
If it's not updated after changing the key, either there is a problem with update that should be fixed, or it's not needed on update so not needed here.

Reply:
I think that I'm missing something with the flow of this